### PR TITLE
Separate command and config services

### PR DESCRIPTION
**Pull request by nagyrobi**

### CODE REVIEW

**Comment by nagyrobi** [comment]
Please have a look at this

**Comment by dgomes** [comment]
I think you should reopen 😬

File: custom_components/openhasp/__init__.py [hunk 1]
```diff
@@ -504,6 +525,22 @@ async def async_change_page(self, page):
         )
         self.async_write_ha_state()
 
+    async def async_command_wrapper(self, keyword, parameters):
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:528]:
```suggestion
    async def async_command_service(self, keyword, parameters):
```

File: custom_components/openhasp/__init__.py [hunk 2]
```diff
@@ -153,7 +159,21 @@ async def async_setup(hass, config):
     component.async_register_entity_service(
         SERVICE_CLEAR_PAGE, {vol.Optional(ATTR_PAGE): int}, "async_clearpage"
     )
+    component.async_register_entity_service(
+        SERVICE_COMMAND, {
+            vol.Required(ATTR_COMMAND_KEYWORD): cv.string, 
+            vol.Optional(ATTR_COMMAND_PARAMETERS, default=""): cv.string
+        }, "async_command_wrapper"
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:166]:
```suggestion
        }, "async_command_service"
```

File: custom_components/openhasp/__init__.py [hunk 3]
```diff
@@ -225,6 +245,7 @@ async def async_unload_entry(hass, entry):
         hass.services.async_remove(DOMAIN, SERVICE_PAGE_CHANGE)
         hass.services.async_remove(DOMAIN, SERVICE_LOAD_PAGE)
         hass.services.async_remove(DOMAIN, SERVICE_CLEAR_PAGE)
+        hass.services.async_remove(DOMAIN, SERVICE_COMMAND)
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:248]: missing remove of config_service

File: custom_components/openhasp/__init__.py [hunk 4]
```diff
@@ -504,6 +525,22 @@ async def async_change_page(self, page):
...
 
+    async def async_command_wrapper(self, keyword, parameters):
+        """Sends commands directly to the plate entity (as a wrapper for MQTT commands sent to hasp/<nodename>/command)"""
+        cmd_topic = f"{self._topic}/command"
+
+        self.hass.components.mqtt.async_publish(
+            cmd_topic, f"{keyword} {parameters}".strip(), qos=0, retain=False
+        )
+
+    async def async_set_config(self, config_topic, parameters):
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:536]:
```suggestion
    async def async_config_service(self, config_topic, parameters):
```

File: custom_components/openhasp/const.py [hunk 5]
```diff
@@ -84,13 +84,19 @@
...
+ATTR_CONFIG_PARAMETERS = "parameters"
 
 SERVICE_WAKEUP = "wakeup"
 SERVICE_CLEAR_PAGE = "clear_page"
 SERVICE_LOAD_PAGE = "load_pages"
 SERVICE_PAGE_CHANGE = "change_page"
 SERVICE_PAGE_NEXT = "next_page"
 SERVICE_PAGE_PREV = "prev_page"
+SERVICE_COMMAND = "command"
+SERVICE_SET_CONFIG = "set_configuration"
```
Comment (dgomes) [inline custom_components/openhasp/const.py:99]:
```suggestion
SERVICE_CONFIG = "config"
```

File: custom_components/openhasp/__init__.py [hunk 6]
```diff
@@ -60,6 +64,8 @@
     SERVICE_PAGE_NEXT,
     SERVICE_PAGE_PREV,
     SERVICE_WAKEUP,
+    SERVICE_COMMAND,
+    SERVICE_SET_CONFIG,
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:68]:
```suggestion
    SERVICE_CONFIG,
```

**Review by dgomes — COMMENTED** [review]
just some tweaking

**Comment by nagyrobi** [comment]
Can you merge this also in the push_image so we can test them together?

**Pull request merged**